### PR TITLE
Read PowerVS images from the release stream

### PR DESCRIPTION
**Pull request by mkumatag**
With recent PR into the installer included the [PowerVS](https://www.ibm.com/in-en/products/power-virtual-server) images into the rhcos.json file, this enhancement will allow us to use the powervs image in the IBM Cloud ppc64le support

Installer PR: https://github.com/openshift/installer/pull/5377

### CODE REVIEW

File: support/releaseinfo/fixtures/4.10-image-references.json [hunk 1]
```diff
@@ -0,0 +1,2507 @@
+{
```
Comment (enxebre) [inline support/releaseinfo/fixtures/4.10-image-references.json:1]: how does this file help to test the new changes?
wouldn't we need to include a 4.10 bootimages.yaml file as we have for 4.8 and then modify `/test/main` `Extract` to account for it?

Comment (mkumatag) [reply support/releaseinfo/fixtures/4.10-image-references.json:1]: I did include the file: `4.10-installer-coreos-bootimages.yaml` not sure if that's the one you are talking about.

Comment (mkumatag) [reply support/releaseinfo/fixtures/4.10-image-references.json:1]: and I see a testcase here: https://github.com/openshift/hypershift/blob/6786c4ec968565c4ebdede2285ebda3343b1ee50/support/releaseinfo/registryclient/test/main.go#L72:L77

I will take a look at it.

Comment (csrwng) [reply support/releaseinfo/fixtures/4.10-image-references.json:1]: That test case is to test the performance of the image extraction code and not necessarily the format of the file.
The new format is getting tested here: https://github.com/openshift/hypershift/pull/690/files#diff-41538f4188cc1a87d29ff10f503cf3dc06cc6838128c9b4280d14514e0f9873dR19
However, I think we could benefit from some assertions that the new platform data is getting included in the parsed result.

Comment (enxebre) [reply support/releaseinfo/fixtures/4.10-image-references.json:1]: right, that's not validating at all that this new code works as expected, see also https://github.com/openshift/hypershift/blob/main/support/releaseinfo/registryclient/test/main.go#L72-L76

Comment (csrwng) [reply support/releaseinfo/fixtures/4.10-image-references.json:1]: Like I mentioned, that's for testing the performance of the image extraction code. I don't think we need to validate other versions there. All we're testing is whether the extracted data is the same as what was expected.

Comment (mkumatag) [reply support/releaseinfo/fixtures/4.10-image-references.json:1]: > However, I think we could benefit from some assertions that the new platform data is getting included in the parsed result.

any specific assertion you are looking for?

Comment (csrwng) [reply support/releaseinfo/fixtures/4.10-image-references.json:1]: Just that you can read the data for the new platform

Comment (mkumatag) [reply support/releaseinfo/fixtures/4.10-image-references.json:1]: @csrwng added the test in the latest commit, ptal

File: support/releaseinfo/releaseinfo_test.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+func TestReleaseInfoPowerVS(t *testing.T) {
+	metadata, err := DeserializeImageMetadata(fixtures.CoreOSBootImagesYAML_4_10)
+	if err != nil {
+		t.Fatal(err)
+	}
+	arch, ok := metadata.Architectures["ppc64le"]
+	if !ok {
+		t.Fatal("metadata does not contain the ppc64le architecture")
+	}
+	if len(arch.Images.PowerVS.Regions) == 0 {
```
Comment (enxebre) [inline support/releaseinfo/releaseinfo_test.go:19]: type CoreOSPowerVSImage struct {
	Release string `json:"release"`
	Object  string `json:"object"`
	Bucket  string `json:"bucket"`
	URL     string `json:"url"`
}

do we know if any of these are being properly populated?

Comment (mkumatag) [reply support/releaseinfo/releaseinfo_test.go:19]: @enxebre amended the latest commit with that additional check as well, ptal

**Comment by enxebre** [comment]
thanks for addressing all comments
/approve

**Comment by mkumatag** [comment]
@csrwng can you please lgtm again?

/retest-required

**Comment by enxebre** [comment]
/hold cancel
/lgtm
as per https://github.com/openshift/hypershift/pull/690#issuecomment-975776362

**Pull request merged**